### PR TITLE
SQL_Task. Slava Telyatko

### PR DESCRIPTION
**Pull request by SlavaTel**
https://travis-ci.org/github/SlavaTel/db-assignments

### CODE REVIEW

File: task/sql-tasks.js [hunk 1]
```diff
@@ -177,7 +273,13 @@ async function task_1_12(db) {
  *
  */
 async function task_1_13(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT
```
Comment (AlexMokin) [inline task/sql-tasks.js]: it's better to write this query like this
notice that from-clause is optional.



```
select
   (select count()...) as 'T'
   (select count()...) as 'TT'
```

File: task/sql-tasks.js [hunk 2]
```diff
@@ -199,7 +309,24 @@ async function task_1_14(db) {
  *
  */
 async function task_1_15(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT
+            SUM(MONTH(OrderDate) = 1) AS "January",
+            SUM(MONTH(OrderDate) = 2) AS "February",
```
Comment (AlexMokin) [inline task/sql-tasks.js]: use count when you need to count records, it shows intent better

File: task/sql-tasks.js [hunk 3]
```diff
@@ -280,7 +465,26 @@ async function task_1_21(db) {
...
+        INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID 
+        INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+        INNER JOIN Products ON OrderDetails.ProductID = Products.ProductID
+        WHERE OrderDetails.UnitPrice =
+        (SELECT
+            MAX(OrderDetails.UnitPrice)
+            FROM Customers AS Customers1
+            INNER JOIN Orders ON Customers1.CustomerID = Orders.CustomerID 
+            INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+            WHERE Customers1.CompanyName = Customers.CompanyName
```
Comment (AlexMokin) [inline task/sql-tasks.js]: company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.

**Review by AlexMokin — COMMENTED** [review]
Looks good. See comments.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/sql-tasks.js (hunk 3 above)
Comment (SlavaTel) [reply task/sql-tasks.js]: Now the request time is about 39ms (144ms was before)

File: task/sql-tasks.js [hunk 4]
```diff
@@ -199,7 +308,24 @@ async function task_1_14(db) {
...
+            (SELECT COUNT(*) FROM Orders WHERE MONTH(OrderDate) = 5 AND YEAR(OrderDate) = 1997) AS "May",
+            (SELECT COUNT(*) FROM Orders WHERE MONTH(OrderDate) = 6 AND YEAR(OrderDate) = 1997) AS "June",
+            (SELECT COUNT(*) FROM Orders WHERE MONTH(OrderDate) = 7 AND YEAR(OrderDate) = 1997) AS "July",
+            (SELECT COUNT(*) FROM Orders WHERE MONTH(OrderDate) = 8 AND YEAR(OrderDate) = 1997) AS "August",
+            (SELECT COUNT(*) FROM Orders WHERE MONTH(OrderDate) = 9 AND YEAR(OrderDate) = 1997) AS "September",
+            (SELECT COUNT(*) FROM Orders WHERE MONTH(OrderDate) = 10 AND YEAR(OrderDate) = 1997) AS "October",
+            (SELECT COUNT(*) FROM Orders WHERE MONTH(OrderDate) = 11 AND YEAR(OrderDate) = 1997) AS "November",
+            (SELECT COUNT(*) FROM Orders WHERE MONTH(OrderDate) = 12 AND YEAR(OrderDate) = 1997) AS "December"
+        FROM Orders
+        GROUP BY 1,2,3,4,5,6,7,8,9,10,11,12;
```
Comment (AlexMokin) [inline task/sql-tasks.js]: it doesn't make sence, you don't need to group anything. Also you don't nedd from-clause it's optional in this case.

But what i mean is to do this

image

**Review by AlexMokin — COMMENTED** [review]
Looks good. See new comment for task_1_15 task.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/sql-tasks.js (hunk 4 above)
Comment (SlavaTel) [reply task/sql-tasks.js]: I think I understand what you mean

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**